src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java

```diff
@@ -290,6 +295,12 @@ public ResponseEntity<Course> updateCourse(@PathVariable Long courseId, @Request
         courseUpdate.setPrerequisites(existingCourse.getPrerequisites());
         courseUpdate.setTutorialGroupsConfiguration(existingCourse.getTutorialGroupsConfiguration());
         courseUpdate.setOnlineCourseConfiguration(existingCourse.getOnlineCourseConfiguration());
+        courseUpdate.setLearningPathsConfiguration(existingCourse.getLearningPathsConfiguration());
+
+        if (courseUpdate.getLearningPathsEnabled() && courseUpdate.getLearningPathsConfiguration() == null) {
+            courseUpdate.setLearningPathsConfiguration(new LearningPathsConfiguration());
```

Comment:
when creating a new LearningPathConfig on course creation one would also not need this conditional